src/lints/manual/helpers.rs

```diff
@@ -363,3 +364,75 @@ pub fn expr_if_get_var_name_and_err(expr_if: AstExprIf, db: &dyn SyntaxGroup) ->
 
     (option_var_name, err)
 }
+
+/// Checks if the expr is a function call of NEVER type
+pub fn is_never_type(expr: &Expr, db: &dyn SemanticGroup, arenas: &Arenas) -> bool {
```

Comment:
Imho this func name isn't verbose enough. I mean, we need to also specify that we are looking at function calls as well blocks that has a tail as a function call